### PR TITLE
go.mod: bump cockroach-go to v2

### PR DESCRIPTION
**Pull request by q3k**
[github.com/cockroachdb/cockroach-go](https://github.com/cockroachdb/cockroach-go) has recently [hit v2](https://github.com/cockroachdb/cockroach-go/pull/78) as part of a gomodization push. No breaking API changes seem to have been introduced, as far as I can tell.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -27,11 +28,10 @@ require (
 	github.com/google/go-github v17.0.0+incompatible
 	github.com/gorilla/mux v1.7.4 // indirect
 	github.com/hashicorp/go-multierror v1.1.0
-	github.com/jackc/pgconn v1.3.2 // indirect
 	github.com/kardianos/osext v0.0.0-20190222173326-2bc1f35cddc0 // indirect
-	github.com/lib/pq v1.3.0
+	github.com/lib/pq v1.4.0
 	github.com/markbates/pkger v0.15.1
-	github.com/mattn/go-sqlite3 v1.10.0
+	github.com/mattn/go-sqlite3 v2.0.1+incompatible
```
Comment (dhui) [inline go.mod:34]: Did you run `go mod tidy`? I don't see this dependency [being used](https://github.com/cockroachdb/cockroach-go/blob/master/go.mod)

Comment (q3k) [reply go.mod:32]: I now have - and also rebased on top of current master. But the dependency is still there, so I'm not sure what else I can do?

Comment (dhui) [reply go.mod:32]: Try checking out just this line and running `go mod tidy`

Comment (dhui) [reply go.mod:32]: Weird I'm having the same issue after only adding `v2` to the cockroach db driver import and running `go mod tidy`, I have the same issue... Not sure why Cockroach DB or one of it's dependencies depend on sqlite...

```shell
$ go mod tidy
go: finding module for package github.com/cockroachdb/cockroach-go/v2/crdb
go: downloading github.com/cockroachdb/cockroach-go/v2 v2.1.0
go: downloading github.com/cockroachdb/cockroach-go v2.0.1+incompatible
go: found github.com/cockroachdb/cockroach-go/v2/crdb in github.com/cockroachdb/cockroach-go/v2 v2.1.0
go: downloading github.com/mattn/go-sqlite3 v2.0.3+incompatible
go: downloading go.uber.org/atomic v1.6.0
go: downloading golang.org/x/crypto v0.0.0-20201016220609-9e8e0b390897
go: downloading golang.org/x/text v0.3.4
go: downloading github.com/shopspring/decimal v0.0.0-20200419222939-1884f454f8ea
go: downloading github.com/mattn/go-colorable v0.1.6
$ git diff go.mod
diff --git a/go.mod b/go.mod
index 5f1aea2..0240eea 100644
--- a/go.mod
+++ b/go.mod
@@ -9,7 +9,7 @@ require (
        github.com/bitly/go-hostpool v0.0.0-20171023180738-a3a6125de932 // indirect
        github.com/bmizerany/assert v0.0.0-20160611221934-b7ed37b82869 // indirect
        github.com/cenkalti/backoff/v4 v4.0.2
-       github.com/cockroachdb/cockroach-go v0.0.0-20190925194419-606b3d062051
+       github.com/cockroachdb/cockroach-go/v2 v2.1.0
        github.com/containerd/containerd v1.4.1 // indirect
        github.com/cznic/mathutil v0.0.0-20180504122225-ca4c9f2c1369 // indirect
        github.com/denisenkom/go-mssqldb v0.0.0-20200620013148-b91950f658ec
@@ -25,12 +25,11 @@ require (
        github.com/google/go-github v17.0.0+incompatible
        github.com/gorilla/mux v1.7.4 // indirect
        github.com/hashicorp/go-multierror v1.1.0
-       github.com/jackc/pgconn v1.3.2 // indirect
        github.com/kardianos/osext v0.0.0-20190222173326-2bc1f35cddc0 // indirect
        github.com/ktrysmt/go-bitbucket v0.6.4
        github.com/lib/pq v1.8.0
        github.com/markbates/pkger v0.15.1
-       github.com/mattn/go-sqlite3 v1.10.0
+       github.com/mattn/go-sqlite3 v2.0.3+incompatible
        github.com/mutecomm/go-sqlcipher/v4 v4.4.0
        github.com/nakagami/firebirdsql v0.0.0-20190310045651-3c02a58cfed8
        github.com/neo4j/neo4j-go-driver v1.8.1-0.20200803113522-b626aa943eba
@@ -43,8 +42,7 @@ require (
        github.com/xdg/stringprep v1.0.0 // indirect
        gitlab.com/nyarla/go-crypt v0.0.0-20160106005555-d9a5dc2b789b // indirect
        go.mongodb.org/mongo-driver v1.1.0
-       go.uber.org/atomic v1.4.0
-       golang.org/x/crypto v0.0.0-20200709230013-948cd5f35899 // indirect
+       go.uber.org/atomic v1.6.0
        golang.org/x/net v0.0.0-20201029221708-28c70e62bb1d // indirect
        golang.org/x/sys v0.0.0-20201029080932-201ba4db2418 // indirect
        golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e // indirect
$
```

Comment (dhui) [reply go.mod:32]: Well... [that](https://github.com/cockroachdb/cockroach-go/blob/master/go.mod) explains it!
Looks like sqlite3 had [some issues](https://github.com/mattn/go-sqlite3#go-sqlite3) with module versioning.
Will hold off on merging this until the upstream dependency issues have been resolved.

**Comment by dhui** [comment]
Closing in favor of #595

**Pull request closed without merging**